burst-kotlin-plugin-tests/src/test/kotlin/app/cash/burst/kotlin/TestInterceptorKotlinPluginTest.kt

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (C) 2025 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.burst.kotlin
+
+import assertk.assertThat
+import assertk.assertions.containsExactly
+import assertk.assertions.isInstanceOf
+import assertk.assertions.isNotNull
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import java.lang.reflect.InvocationTargetException
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi
+
+@OptIn(ExperimentalCompilerApi::class)
+class TestInterceptorKotlinPluginTest {
```

Comment:
Seems like a lot of duplication here and the Gradle plugin tests. What's the responsibility split? These ones should probably test the nuances of behavior (multiple interceptors, order, exceptions, etc.) and the Gradle plugin tests mostly just that integration (runtime automatically added, compiler plugin gets applied and runs, etc.)

Reply:
Yeah, agreed. I moved most of these to the Kotlin plugin test, keeping the JUnit-specific ones in the Gradle plugin test.